clients/fossid-webapp/src/main/kotlin/Extensions.kt

```diff
@@ -302,3 +309,54 @@ suspend fun FossIdRestService.createIgnoreRule(
             "apply_to" to scope.name.lowercase()
         )
     )
+
+suspend fun FossIdRestService.generateReport(
+    user: String,
+    apiKey: String,
+    scanCode: String,
+    reportType: ReportType,
+    selectionType: SelectionType
+): File {
+    val response = generateReport(
+        PostRequestBody(
+            "generate_report",
+            SCAN_GROUP,
+            user,
+            apiKey,
+            "scan_code" to scanCode,
+            "report_type" to reportType.name.lowercase(),
+            "selection_type" to selectionType.name.lowercase()
+        )
+    )
+
+    require(response.isSuccessful) {
+        """ ${response.code()} - ${response.message()}
+            ${response.errorBody()?.string()}
+        """.trimIndent()
+    }
+
+    // The API is quirky here: if the report type is HTML, the result is returned as a standard HTML response, without
+    // filename. For any other report types, the result is returned as an attachment with a filename in the headers
+    // (even for report types generating HTML such as DYNAMIC).
+    val fileName = if (reportType == ReportType.HTML) "${UUID.randomUUID()}-report.html"
+    else {
+        response.headers()["Content-disposition"]?.split(';')?.firstNotNullOfOrNull {
+            it.trim().withoutPrefix("filename=")?.removeSurrounding("\"")
+        } ?: throw IllegalStateException("Cannot determine name of the report")
+    }
+
+    val tempDir = createTempDirectory("fossid_report").toFile()
+        .apply { deleteOnExit() }
```

Comment:
That's dirty. IMO we should rather add a parameter for the parent directory to this function.